bindings/azure/servicebusqueues/servicebusqueues.go

```diff
@@ -267,6 +257,48 @@ func (a *AzureServiceBusQueues) Read(subscribeCtx context.Context, handler bindi
 	return nil
 }
 
+// getSender returns the Sender object, creating a new connection if needed
+func (a *AzureServiceBusQueues) getSender() (*servicebus.Sender, error) {
+	// Check if the sender already exists
+	a.senderLock.RLock()
+	if a.sender != nil {
+		a.senderLock.RUnlock()
+		return a.sender, nil
+	}
+	a.senderLock.RUnlock()
+
+	// Acquire a write lock then try checking a.sender again in case another goroutine modified that in the meanwhile
+	a.senderLock.Lock()
+
+	if a.sender != nil {
+		a.senderLock.Unlock()
+		return a.sender, nil
+	}
+
+	// Create a new sender
+	sender, err := a.client.NewSender(a.metadata.QueueName, nil)
+	if err != nil {
+		a.senderLock.Unlock()
+		return nil, err
+	}
+	a.sender = sender
+	a.senderLock.Unlock()
+
+	return sender, nil
+}
+
+// deleteSender deletes the sender, closing the connection
+func (a *AzureServiceBusQueues) deleteSender() {
+	a.senderLock.Lock()
+	if a.sender != nil {
+		closeCtx, closeCancel := context.WithTimeout(context.Background(), time.Second)
+		_ = a.sender.Close(closeCtx)
+		closeCancel()
```

Comment:
did you consider close it in background?

Given the fact that you're ignoring the error and the timeout is actually 1s (so we can have 1s lock here?)

Reply:
A timeout is still needed or we could leak a goroutine if the connection just hanged forever

Reply:
yes, you can keep the timeout being 1s though

Reply:
Oh I misunderstood. I thought you meant just using `context.Background()` with no timeout.

I think this needs to happen in foreground. Reason is that after this method returns, the caller usually creates a new connection. It may be best to make sure the old one is actually closed before creating a new one.

Reply:
if you want to keep it in that way, so ideally you should receive a parent context and compose with the 1s timeout,

What I'm trying to say is that the `deleteSender` function can block for a second arbitrarily and its caller has no option to prevent this from happening

Reply:
> What I'm trying to say is that the deleteSender function can block for a second arbitrarily and its caller has no option to prevent this from happening

Yup, that's by design. So there's some slowdown